### ISSUE TITLE
Grpc-Java best practices

### ISSUE THREAD

**Issue by biran0079** [grpc/grpc-java#2338]
Developers may run into similar issues when started using Grpc. Solutions to most of the issue are  scattered around the mailing list and github issues, like custom server executor, stream rcp flow control, server/client request intercepting, error handling, ssl, testing, etc.
I think it would be valuable to document the best practices for Grpc-Java.

**Comment by carl-mastrangelo** [comment]
@dapengzhang0 Is working on making some examples that show off best practices.  See:  https://github.com/grpc/grpc-java/pull/2353

**Comment by hsaliak** [comment]
@carl-mastrangelo is this issue fixed by #2353 ?

**Comment by jjinux** [comment]
Yeah, this is really hard, and #2353 doesn't see to have anything to do with it :-/
